plugins/catalog-backend/src/ingestion/processors/UrlReaderProcessor.ts

```diff
@@ -22,42 +22,77 @@ import { Logger } from 'winston';
 import * as result from './results';
 import {
   CatalogProcessor,
+  CatalogProcessorCache,
   CatalogProcessorEmit,
+  CatalogProcessorEntityResult,
   CatalogProcessorParser,
+  CatalogProcessorResult,
 } from './types';
 
+const CACHE_KEY = 'v1';
+
 type Options = {
   reader: UrlReader;
   logger: Logger;
 };
 
+type CacheItem = {
+  etag: string;
+  value: CatalogProcessorEntityResult[];
+};
+
 export class UrlReaderProcessor implements CatalogProcessor {
   constructor(private readonly options: Options) {}
 
+  getProcessorName() {
+    return 'url-reader';
+  }
+
   async readLocation(
     location: LocationSpec,
     optional: boolean,
     emit: CatalogProcessorEmit,
     parser: CatalogProcessorParser,
+    cache: CatalogProcessorCache,
   ): Promise<boolean> {
     if (location.type !== 'url') {
       return false;
     }
 
+    const cacheItem = await cache.get<CacheItem>(CACHE_KEY);
```

Comment:
wait, how on earth can the location not be part of the cache key? this processor will be used for the reading of all locations everywhere right?

Reply:
It's local to the location entity, but this still needs fixing. The problem right now is that a location entity can point to multiple locations, but the `readLocation` abstracts that away. We could perhaps start the migration away from `readLocation` as part of this, and not provide access to the cache in `readLocation` at all?

Reply:
Sounds to me like just putting the location into the key is the logical thing to do here

Reply:
It's still confusing behaviour because the `readLocation` implementation needs further keying but the `pre/postProcess` do not. We could key by the location here for now if we don't wanna do that rework yet, but it's pretty ugly and I think it's best we get rid of it long term. We will end up needing keying by location in the end though, but it'd be nice if the need was more explicit.

Reply:
We've ended up handling this by supplying a cache implementation that's scoped to each `readLocation` call